test/Conversion/VerifToSMT/verif-to-smt.mlir

```diff
@@ -60,3 +61,89 @@ func.func @test(%arg0: !smt.bv<1>) -> (i1, i1, i1) {
+// CHECK:      }
+// CHECK:      [[XORI:%.+]] = arith.xori [[FOR]]#4, [[TRUE]]
+// CHECK:      smt.yield [[XORI]]
+// CHECK:    }
+// CHECK:    return [[BMC]]
+// CHECK:  }
+// CHECK-LABEL:  func.func @bmc_init() -> (!smt.bv<1>, !smt.bv<1>) {
+// CHECK:    [[FALSE:%.+]] = hw.constant false
+// CHECK:    [[TOCLOCK:%.+]] = seq.to_clock %false
+// CHECK:    [[C0:%.+]] = builtin.unrealized_conversion_cast [[TOCLOCK]] : !seq.clock to !smt.bv<1>
+// CHECK:    [[C1:%.+]] = builtin.unrealized_conversion_cast [[FALSE]] : i1 to !smt.bv<1>
+// CHECK:    return [[C0]], [[C1]]
+// CHECK:  }
+// CHECK:  func.func @bmc_loop([[ARGO:%.+]]: !smt.bv<1>, [[ARG1:%.+]]: !smt.bv<1>)
+// CHECK:    [[C2:%.+]] = builtin.unrealized_conversion_cast [[ARG1]] : !smt.bv<1> to i1
+// CHECK:    [[C3:%.+]] = builtin.unrealized_conversion_cast [[ARG0]] : !smt.bv<1> to !seq.clock
+// CHECK:    [[FROMCLOCK:%.+]] = seq.from_clock [[C3]]
+// CHECK:    [[TRUE]] = hw.constant true
+// CHECK:    [[NCLOCK:%.+]] = comb.xor [[FROMCLOCK]], [[TRUE]]
+// CHECK:    [[NARG:%.+]] = comb.xor [[C2]], [[TRUE]]
+// CHECK:    [[TOCLOCK:%.+]] = seq.to_clock [[NCLOCK]]
+// CHECK:    [[C4:%.+]] = builtin.unrealized_conversion_cast [[TOCLOCK]] : !seq.clock to !smt.bv<1>
+// CHECK:    [[C5:%.+]] = builtin.unrealized_conversion_cast [[NARG]] : i1 to !smt.bv<1>
+// CHECK:    return [[C4]], [[C5]]
+// CHECK:  }
+// CHECK:  func.func @bmc_circuit([[ARGO:%.+]]: !smt.bv<1>, [[ARG1:%.+]]: !smt.bv<32>, [[ARG2:%.+]]: !smt.bv<32>)
+// CHECK:    [[C6:%.+]] = builtin.unrealized_conversion_cast [[ARG2]] : !smt.bv<32> to i32
+// CHECK:    [[C7:%.+]] = builtin.unrealized_conversion_cast [[ARG1]] : !smt.bv<32> to i32
+// CHECK:    [[CN1_I32:%.+]] = hw.constant -1 : i32
+// CHECK:    [[ADD:%.+]] = comb.add [[C7]], [[C6]]
+// CHECK:    [[XOR:%.+]] = comb.xor [[C6]], [[CN1_I32]]
+// CHECK:    [[C9:%.+]] = builtin.unrealized_conversion_cast [[XOR]] : i32 to !smt.bv<32>
+// CHECK:    [[C10:%.+]] = builtin.unrealized_conversion_cast [[ADD]] : i32 to !smt.bv<32>
+// CHECK:    return [[C9]], [[C10]]
+// CHECK:  }
+
+func.func @test_bmc() -> (i1) {
+  %bmc = verif.bmc bound 10 num_regs 1
+  init {
+    %c0_i1 = hw.constant 0 : i1
+    %clk = seq.to_clock %c0_i1
+    verif.yield %clk, %c0_i1 : !seq.clock, i1
+  }
+  loop {
+    ^bb0(%clk: !seq.clock, %stateArg: i1):
+    %from_clock = seq.from_clock %clk
+    %c-1_i1 = hw.constant -1 : i1
+    %neg_clock = comb.xor %from_clock, %c-1_i1 : i1
+    %newStateArg = comb.xor %stateArg, %c-1_i1 : i1
+    %newclk = seq.to_clock %neg_clock
+    verif.yield %newclk, %newStateArg : !seq.clock, i1
+  }
+  circuit {
+  ^bb0(%clk: !seq.clock, %arg0: i32, %state0: i32):
+    %c-1_i32 = hw.constant -1 : i32
+    %0 = comb.add %arg0, %state0 : i32
+    // %state0 is the result of a seq.compreg taking %0 as input
+    %2 = comb.xor %state0, %c-1_i32 : i32
+    verif.yield %2, %0 : i32, i32
+  }
+  func.return %bmc : i1
+}
```

Comment:
A test for `verif.assume` seems to be missing

Reply:
Good catch, thanks!